splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1367,6 +1367,14 @@ public void setMetadataAccessRestrictionEnabled() {
 
     @Override
     public void updateSystemSchemasView(TransactionController tc) throws StandardException {
+        boolean toUpgrade = Boolean.TRUE.equals(EngineLifecycleService.toUpgrade.get());
+        // Only master can upgrade
```

Comment:
I feel like the comment is a bit misleading, there's nothing from here which indicates that only master can upgrade. Only the ones with `toUpgrade` set can upgrade, and it just "so" happens that we set `toUpgrade` with `isMaster`. Now keep in mind that I don't have the whole picture just yet. :)